sherpa/bin/conv_emformer_transducer_stateless2/streaming_server.py

```diff
@@ -347,33 +242,28 @@ def __init__(
         self.log_eps = math.log(1e-10)
 
         self.initial_states = self.model.get_encoder_init_states()
-        self.decoding_method = decoding_method
-
-        assert self.decoding_method in ["greedy_search", "fast_beam_search"]
-
-        self.initial_decoder_out = None
-        self.decoding_graph = None
 
         if decoding_method == "fast_beam_search":
-            self.rnnt_decoding_config = k2.RnntDecodingConfig(
-                vocab_size=self.vocab_size,
-                decoder_history_len=self.context_size,
-                beam=beam,
-                max_states=max_states,
-                max_contexts=max_contexts,
+            self.beam_search = FastBeamSearch(
+                self.vocab_size,
```

Comment:
I suggest using keyword arguments to invoke it as there are many arguments here.

Reply:
I am not sure to understand. Do you mean ```vocab_size=self.vocab_size``` ?

Reply:
> I am not sure to understand. Do you mean `vocab_size=self.vocab_size` ?

Exactly.

Since the constructor has many arguments, using keyword arguments here can reduce the chance of mixing
the orders of arguments while invoking it.

Reply:
Sounds good !
I will finish up everything beginning next week